tests/integration/telemetry/api/accesslogs_test.go

```diff
@@ -107,11 +118,21 @@ defaultProviders:
 `
 				ist := *(common.GetIstioInstance())
 				ist.PatchMeshConfigOrFail(t, t, cfg)
-				runAccessLogsTests(t, true)
+				runAccessLogsTests(t, true, false)
 			})
 		})
 }
 
+func applyTelemetryResourceWithTargetRef(t framework.TestContext, enableLogs bool) {
+	args := map[string]any{
+		"To":                common.GetTarget().(echo.Instances),
+		"TargetGatewayName": common.GetTarget().(echo.Instances).ServiceName() + "-gateway",
+		"DisableLogs":       !enableLogs,
+	}
+	t.ConfigIstio().EvalFile(common.GetAppNamespace().Name(), args, "./testdata/gateway-api.yaml").ApplyOrFail(t)
```

Comment:
You'll probably run into a race condition here since it can take 10-30 seconds for the actual deployment + service to be deployed and for the load balancer IP to be assigned. It's probably better to run this at the very beginning of your targetRef SubTests on L60 and L70